soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4745,9 +4741,90 @@ void DrawRandoEditor(bool& open) {
     ImGui::End();
     }*/
 
+typedef struct {
+    GetItemID giid;
+    ItemID iid;
+    std::string english;
+    std::string german;
+    std::string french;
+} GetItemMessage;
+
+void CreateGetItemMessages(std::vector<GetItemMessage> messageEntries) {
+    CustomMessageManager* customMessage = CustomMessageManager::Instance;
+    customMessage->AddCustomMessageTable(Randomizer::getItemMessageTableID);
+    for (GetItemMessage messageEntry : messageEntries) {
+        customMessage->CreateGetItemMessage(Randomizer::getItemMessageTableID, messageEntry.giid, messageEntry.iid,
+                                            { TEXTBOX_TYPE_BLUE, TEXTBOX_POS_BOTTOM,
+                                              messageEntry.english, messageEntry.german,
+                                              messageEntry.french });
+    }
+}
+
+void CreateScrubMessages() {
+    CustomMessageManager* customMessage = CustomMessageManager::Instance;
+    customMessage->AddCustomMessageTable(Randomizer::scrubMessageTableID);
+    const std::vector<u8> prices = { 10, 40 };
+    for (u8 price : prices) {
+        customMessage->CreateMessage(Randomizer::scrubMessageTableID, price,
+            { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+              "\x12\x38\x82\All right! You win! In return for&sparing me, I will sell you a&%gmysterious item%w!&%r" +
```

Comment:
is there any way we could remove the `\x12\x38\x82\` stuff from these? doesn't need to be a part of this PR but would be nice to have those represented in a more understandable manner

Reply:
Yeah at some point we need to branch out some kind of standardized markup language of some kind for replacing all the control codes. That control code is for playing the sound effect, which currently the only custom messages I needed to add to the tables that did that were the Deku Scrubs. So far for that markup-language style thing I borrowed what you already had for the hint text in spoiler files (so & for newline, ^ to wait for input, and @ for the player name. Then I added a %<letter> syntax for changing colors. I proposed a ${<iconID>} syntax for item icons on the left side of the textbox, maybe we can do a #{<soundID>} syntax for sound effects? Or is there something else you'd like to see?

I could also probably just expose some helper methods and people can do string concatenation as an option at least, but I don't want that to be the only way, since people couldn't really use that if we wanted to allow pulling the strings from an external file later on.

Reply:
might be worth making an issue for updating our custom text syntax to support icons and sounds, that can be a different PR (especially since i'd like to hear some more opinions on syntax)

also, it raises some other questions (does `&/^` make sense? should we use `%` for colors? can we standardize a bit more without making everything too verbose?)

Reply:
We could also potentially use the Non-JP Dev Files table from here https://wiki.cloudmodding.com/oot/Text_Format#Control_Codes unless that could cause copyright issues. It doesn't have any syntax for line break or changing colors though, so we'd need to come up with something else for those.

Reply:
i feel like having our own syntax might be better, especially when thinking about iconIds (they aren't very human readable)

i feel like ideally we'd be able to read a custom text string and understand what it's doing without needing to reference external id tables etc. (trying to balance that with keeping the strings short will be a bit of a challenge, but i'm sure we'll figure something out)

i had an idea for using emoji for text colors (not sure how i feel about using emoji in json but in some ways `🟥` feels better to me than `TEXT_COLOR_RED` or `%r`), so there are definitely some "thinking out of the box" options we can look into